- Implemented calculation sum with `ExecutorService`

Likh-Alex opened the pull request:
- Implemented calculation sum with `ForkJoin`
- Written simple tests

Likh-Alex committed:
- Implemented calculation sum with `ExecutorService`
- Implemented calculation sum with `ForkJoin`
- Written simple tests

Likh-Alex committed:
- replaced `mapToInt` with `reduce` in `getCurrentSum`

Likh-Alex committed:
- extracted adding `partitions` into `List` into `addPartitionsToThread` method

Ketch92 commented on src/main/java/processing/SumListWithCallable.java:
```diff
@@ -0,0 +1,53 @@
+        return summary;
+    }
+
+    private void addPartitionsToThread(List<List<Integer>> partitions,
+                                       List<CallableThread> callableThreads) {
+        for (List<Integer> partition : partitions) {
+            callableThreads.add(new CallableThread(partition));
+        }
+    }
```
Ти можеш замінити весь цей метод на один стрім

Likh-Alex replied:
заменил - выглядит короче) Спасибо

Ketch92 reviewed (approved):
Well done

Likh-Alex committed:
- refactored creation of `callableThreads` with `stream`

Sofasmile commented on src/main/java/thread/CallableThread.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+import java.util.concurrent.Callable;
+
+public class CallableThread implements Callable<Integer> {
```
```suggestion
public class SumCalculatorCallable implements Callable<Integer> {
```

Likh-Alex replied:
Done, thank you

Sofasmile commented on src/main/java/thread/RecursiveTaskThread.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+import java.util.concurrent.RecursiveTask;
+
+public class RecursiveTaskThread extends RecursiveTask<Integer> {
```
```suggestion
public class SumCalculatorRecursiveTask extends RecursiveTask<Integer> {
```

Likh-Alex replied:
Done, thank you

Sofasmile commented on src/main/java/util/Util.java:
```diff
@@ -0,0 +1,24 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Util {
```
```suggestion
public class ListOfNumbersCreator/ListOfNumbersGenerator {
```

Likh-Alex replied:
Done

Sofasmile commented on src/main/java/util/Util.java:
```diff
@@ -0,0 +1,24 @@
+        return list;
+    }
+
+    public int getCurrentSum() {
```
цей метод ми використовуємо лише в тестах, тому пропоную винести його з цього класу туди

Likh-Alex replied:
moved this method to the tests

Sofasmile commented on src/main/java/util/Util.java:
```diff
@@ -0,0 +1,24 @@
+
+public class Util {
+    private static final int DESIRED_SIZE = 1_000_000;
+    private final List<Integer> list = new ArrayList<>();
```
спробуй перенести це поле в метод, де це потрібно. Тоді наш клас буде stateless

Likh-Alex replied:
Перенёс везде где вызывается `getRandomList` и передаю в него же

Sofasmile commented on src/test/java/SumListWithCallableTest.java:
```diff
@@ -0,0 +1,27 @@
+import util.Util;
+import java.util.List;
+
+import static org.junit.Assert.*;
```
автоімпорти злоо - старайся їх не використовувати

Likh-Alex replied:
есть за мной такой грешок, поправил везде где был автоимпорт со звёздочкой

Sofasmile commented on src/test/java/SumListWithCallableTest.java:
```diff
@@ -0,0 +1,27 @@
+import static org.junit.Assert.*;
+
+public class SumListWithCallableTest {
+    private static final int RANDOM_QUALIFIER = 3;
+    private static SumListWithCallable callableSum;
+    private static Util util;
```
чому ці змінні статичні?

Likh-Alex replied:
Потому что я забыл их правильно объявить) Убрал

Sofasmile commented on src/test/java/SumListWithCallableTest.java:
```diff
@@ -0,0 +1,27 @@
+        int expected = util.getCurrentSum();
+        assertEquals(expected, actual);
+    }
+}
```
пуста строка

Likh-Alex replied:
добавил... -_-

Sofasmile commented on src/test/java/SumListWithCallableTest.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Test
+    public void getSummary_Ok() {
```
загалом одного тесту замало, але для даного завдання можеш залишити як є

Likh-Alex replied:
Понял, спасибо)

Sofasmile commented on src/test/java/SumListWithForkJoinTest.java:
```diff
@@ -0,0 +1,27 @@
+        int expected = util.getCurrentSum();
+        assertEquals(expected, actual);
+    }
+}
```
пуста строка

Likh-Alex replied:
Теперь есть)

Likh-Alex committed:
- refactoring of Class names
- moved `getCurrentSum` to tests
- moved `LIST_SIZE` where it is used with `getCurrentSum`
- added empty lines where required
